FParsec/Emit.fs

```diff
@@ -22,16 +22,23 @@ open FParsec.Range
 
 let mutable private assemblyBuilder = null
 let mutable private moduleBuilder = null
-let private createTypeBuilderSyncRoot = new obj()
+let private createTypeBuilderSyncRoot = obj()
+
+let getModuleBuilder() =
+    match moduleBuilder with
```

Comment:
I don't think this access is safe. In C# you'd need to make the `moduleBuilder` variable volatile in order to ensure the variable is loaded with acquire semantics. This also seems to match the [draft memory model documentation](https://github.com/dotnet/runtime/pull/75790/files).

Reply:
[There is no `volatile` in the memory document example.](https://github.com/dotnet/runtime/pull/75790/files#diff-34c8358c0bb94c612b0d6d189eb597202919dc4785920041081583b87981287eR208-R234) `lock` places its own barriers.

Reply:
Maybe that's an oversight or error. I've haven't seen an argument why this is safe in general. The read outside the lock doesn't use acquire semantics and isn't synchronized with the lock operations.

Reply:
I eventually just used `lazy`. 😁